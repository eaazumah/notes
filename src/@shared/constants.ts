import {Dimensions, StatusBarStyle} from 'react-native';

export const colors = {
  grey: '#E1E0DF',
  white: '#FFFFFF',
  success: '#5cb85c',
  danger: '#d9534f',
  black: '#242A37',
  primary: '#3269ff',
  secondary: '#FF00FF',
};

export const lightTheme = {
  primary: '#3269ff',
  secondary: '#F25F5C',
  primaryText: '#242A37',
  secondaryText: '#E1E0DF',
  backgroundColor: '#F5F5F5',
  surface: '#e6e6e6',
  barStyle: 'dark-content' as StatusBarStyle,
  tabBarOptions: {
    tabBarActiveTintColor: '#3269ff',
    // tabBarInactiveTintColor: 'grey',
    tabBarStyle: {
      backgroundColor: '#FFFFFF',
    },
  },
};

export const darkTheme = {
  primary: '#3269ff',
  secondary: '#F25F5C',
  primaryText: '#FFFFFF',
  secondaryText: '#E1E0DF',
  backgroundColor: '#151515',
  surface: '#171c26',
  barStyle: 'light-content' as StatusBarStyle,
  tabBarOptions: {
    tabBarActiveTintColor: '#3269ff',
    // tabBarInactiveTintColor: '#E1E0DF',
    tabBarStyle: {
      backgroundColor: '#151515',
    },
  },
};

export const routes = {
  home: 'Home',
  editor: 'Editor',
  splash: 'Splash',
  archived: 'Archived',
  favorites: 'Favorites',
  bottomTab: 'BottomTab',
};

export const themes = {
  dark: darkTheme,
  light: lightTheme,
};

export const deviceWidth = Dimensions.get('screen').width;
export const deviceHeight = Dimensions.get('screen').height;

export const pattern = {
  italic: '*&*&*',
  bold: '#@#@',
};

export const regex = {
  half: /\*\&\*\&\*|\#\@\#\@/,
  halfItalic: /\*\&\*\&\*/,
  italic: /\*\&\*\&\*(.*)\*\&\*\&\*/,
  italicEmpty: /\*\&\*\&\*\*\&\*\&\*/g,

  halfBold: /\#\@\#\@/,
  bold: /\#\@\#\@(.*)\#\@\#\@/,
  boldEmpty: /\#\@\#\@\#\@\#\@/g,
  full: /\*\&\*\&\*(.*)\*\&\*\&\*|\#\@\#\@(.*)\#\@\#\@/,
};

const text = `*&*&*italic*&*&*   #@#@bold#@#@`;
