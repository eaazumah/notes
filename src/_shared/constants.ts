import {StatusBarStyle} from 'react-native';

export const colors = {
  grey: '#E1E0DF',
  white: '#FFFFFF',
  success: '#5cb85c',
  danger: '#d9534f',
  black: '#242A37',
  primary: '#FF7E00',
  secondary: '#FF00FF',
};

export const lightTheme = {
  black: '#242A37',
  primary: '#FF8C29',
  secondary: '#F25F5C',
  primaryText: '#242A37',
  secondaryText: '#E1E0DF',
  success: '#5cb85c',
  danger: '#d9534f',
  backgroundColor: '#F5F5F5',
  surface: '#E1E0DF',
  grey: '#E1E0DF',
  barStyle: 'dark-content' as StatusBarStyle,
  white: '#FFFFFF',
  tabBarOptions: {
    inactiveTintColor: 'grey',
    activeTintColor: '#FF8C29',
    style: {
      backgroundColor: '#FFFFFF',
    },
  },
};

export const darkTheme = {
  black: '#242A37',
  primary: '#FF8C29',
  secondary: '#F25F5C',
  primaryText: '#FFFFFF',
  secondaryText: '#E1E0DF',
  success: '#5cb85c',
  danger: '#d9534f',
  motorbike: '#0B3954',
  backgroundColor: '#151515',
  surface: '#272727',
  grey: '#E1E0DF',
  barStyle: 'light-content' as StatusBarStyle,
  white: '#FFFFFF',
  tabBarOptions: {
    inactiveTintColor: '#E1E0DF',
    activeTintColor: '#FF8C29',
    style: {
      backgroundColor: '#151515',
    },
  },
};

export const routes = {
  home: 'home',
  editor: 'editor',
  splash: 'splash',
  archived: 'archived',
  favorites: 'favorites',
};
