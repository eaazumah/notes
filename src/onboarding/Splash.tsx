import React, {useEffect} from 'react';
import Logo from '../@shared/assets/logo.svg';
import {navigateToBottomTab} from '../@shared/services/navigator';
import Container from '../@shared/styled/Container';

interface Props {}

const Splash: React.FC<Props> = () => {
  useEffect(() => {
    setTimeout(() => {
      navigateToBottomTab();
    }, 1000);
  }, []);

  return (
    <Container>
      <Logo width={200} height={200} />
    </Container>
  );
};

export default Splash;
