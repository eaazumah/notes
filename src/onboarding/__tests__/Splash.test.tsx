import 'jest-styled-components';
import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {lightTheme} from '../../@shared/constants';
import Splash from '../Splash';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <Splash />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
