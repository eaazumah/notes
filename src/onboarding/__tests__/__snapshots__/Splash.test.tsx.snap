// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <SvgMock
    height={200}
    width={200}
  />
</View>
`;
